http-clients/src/test/java/org/triplea/http/client/web/socket/WebSocketConnectionTest.java

```diff
@@ -70,11 +71,32 @@ void setup() {
 
     @Test
     void connect() throws Exception {
+      when(webSocketClient.connectBlocking(anyLong(), any())).thenReturn(true);
       webSocketConnection.connect();
 
       verify(webSocketClient).connectBlocking(anyLong(), any());
     }
 
+    @Test
+    void connectFails() {
+      webSocketConnection.setClient(
+          new WebSocketClient(URI.create("wss://server.invalid")) {
```

Comment:
Does this fail quickly, or is there a timeout? I wonder in regards to test runtime.

On a minor note, it took some reverse engineering to figure out why we expect a RuntimeException. Perhaps that could be fixed by adding a `@DisplayName` annotation, eg:
```
@DisplayName("Verify connect fails with exception when connecting to a non-existent endpoint");
```

Reply:
I just tested this again, it takes slightly longer than normal unit test. This test case takes about 5 seconds to execute. I'll try to inject a mock socket factory to speed things up, the tradeoff is that we might end up being further away from reality

Reply:
Thanks for checking, have you considered the trade-off of toggling the timeout to be something very short, like 5ms?

Reply:
Considered that, but by looking at the code it seems that the timeout is already 0?
Not sure if this is the value for "default timeout" or something but I haven't actually tried using 1

Reply:
I'm a bit confused, I was expecting this chunk of code to control the connectoin timeout:
```
    final Result<Boolean> connectionAttempt =
        Interruptibles.awaitResult(
            () -> client.connectBlocking(DEFAULT_CONNECT_TIMEOUT_MILLIS, TimeUnit.MILLISECONDS));
```

My thinking is we could go back to connect timeout to be parameterized so that test could control it and then inject a very short timeout value. Thoughts? I'm also curious where you are spotting the timeout of '0'.

Reply:
Ah, of course. I'll blame it on the fact that I'm pretty tired right now ^^
The 0 timeout actually represents the internal timeout of WebSocketClient. I had to dig into the source code to be able to make the test work, so that's where I found it.

Anyways I feel like this is going back and forth, which is why I'd suggest to merge this PR now, so you can try to find a solution that 1. works and 2. is better in your opinion.
I also think that making the external timeout/our custom timeout super short only tests a different scenario where the connection actually takes too long to establish (maybe caused by high packet loss) vs a non-responding server.

I also tried using a single WebSocketClient mock for all the test cases to keep things simpler, however it seems like this actually achieved the opposite for this one test case.

In any case I'll recommend you fiddle around with the test code a little. I'd say it's in a fairly decent shape right now, but I agree that it can be improved, I'm just not sure how exactly it should be improved.

Reply:
Sounds good :+1: